### DISCUSSION TITLE
In 1.10 ClusterInfo.cluster is now ICluster as opposed to Cluster. How to get Cluster from that?

### DISCUSSION THREAD

**Discussion by shapirov103** [aws-quickstart/cdk-eks-blueprints#760] [General]
`ICluster` refactoring was performed to address cluster import. For imported clusters, the type is ICluster and does not have full function support that is available for newly provisioned clusters. For example, cluster authentication is not accessible for imported clusters. 

If you are provisioning and maintaining clusters with the blueprints, then you can always rely on the type to be `Cluster`. 

Example checking for the type of cluster:

```typescript
if(!(clusterInfo.cluster instanceof Cluster)) {
  logger.warn(`Some cluster updates (such as access management) are not supported with imported clusters` );
  return;
}
```

Example casting to `Cluster`:

```typescript
 const eksCluster: Cluster = clusterInfo.cluster;
// do something with eksCluster e.g.:
 eksCluster.awsAuth.addMastersRole(...);
```

**Comment by shapirov103** [comment]
An example of an explicit cast from ICluster to cluster:

```typescript
import * as eks from 'aws-cdk-lib/aws-eks';
...
const eksCluster = clusterInfo.cluster as eks.Cluster;
```

**Comment by vumdao** [comment]
Previously, when `ClusterInfo.cluster` was `Cluster` I could use `addCdk8sChart` to apply manifest but now it does nothing. Do you have any idea @shapirov103 ?

**Reply by shapirov103** [reply]
@vumdao please describe the current behavior when invoking the `addCdk8sChart` method as it is expect to return the `KubernetesManifest`. "Does nothing" - returns null? No errors?

If you provide an example (inline or on git), happy to give it a shot. If a failure is coming from CDK, we can raise an issue there.
